services/email/app/main.py

```diff
@@ -78,77 +78,21 @@ def send_report():
 
     logger.info(f'successfully read run {run_id}.')
 
-    statuses = defaultdict(lambda: 0)
-    results = defaultdict(lambda: 0)
-
-    failure = []
-
-    logging.info('begin composing report')
-    for task in tasks:
-        status = task['status']
-        result = task['result']
-
-        statuses[status] = statuses[status] + 1
-        results[result] = results[result] + 1
-
-        if result != 'Passed':
-            if run['details'].get('a01.reserved.product', None) == 'azurecli':
-                test_name = task['settings']['classifier']['identifier']
-                if test_name.startswith('azure.cli.command_modules'):
-                    module_name = test_name.split('.')[3]
-                else:
-                    module_name = 'CORE'
-            else:
-                module_name = None
-
-            failure.append(
-                (task['id'],
-                 task['name'].rsplit('.')[-1],
-                 task['status'],
-                 task['result'],
-                 (task.get('result_details') or dict()).get('duration'),
-                 module_name))
-
-    status_summary = ' | '.join([f'{status_name}: {count}' for status_name, count in statuses.items()])
-    result_summary = ' | '.join([f'{result or "Not run"}: {count}' for result, count in results.items()])
-
-    creation = datetime.strptime(run['creation'], '%Y-%m-%dT%H:%M:%SZ') - timedelta(hours=8)
-
-    summaries = [('Id', run['id']),
-                 ('Creation', str(creation) + ' PST'),
-                 ('Creator', run['details'].get('a01.reserved.creator', 'N/A')),
-                 ('Remark', run['settings'].get('a01.reserved.remark', 'N/A')),
-                 ('Live', run['settings'].get('a01.reserved.livemode')),
-                 ('Task', status_summary),
-                 ('Image', run['settings'].get('a01.reserved.imagename', 'N/A')),
-                 ('Result', result_summary)]
-
-    content = f"""\
-    <html>
-        <body>
-            <div>
-                <h2>Summary</h2>
-                {tabulate(summaries, tablefmt="html")}
-            </div>
-            <div>
-                <h2>Failures</h2>
-                {tabulate(failure, headers=("id", "name", "status", "result", "duration(ms)", "module"), tablefmt="html")}
-            </div>
-            <div>
-                <h2>More details</h2>
-                <p>Install the latest release of A01 client to download log and recordings.</p>
-                <p>Instruction is here: https://github.com/azure/adx-automation-client</p>
-                <code>
-                $ a01 login<br>
-                $ a01 get runs -l {run['id']}<br>
-                </code>
-                <p>Contact: trdai@microsoft.com</p>
-            </div>
-        </body>
-    </html>"""
+    product = run['details'].get('a01.reserved.product', 'azurecli')
```

Comment:
Let's not fallback on `azurecli`. If a run doesn't have product name it should fail.

Reply:
Fixed